src/editor/HostDialog.ui

```diff
@@ -16,28 +16,86 @@
   <layout class="QVBoxLayout" name="verticalLayout">
    <item>
     <widget class="QGroupBox" name="openProjectGroup">
+     <property name="sizePolicy">
+      <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
+       <horstretch>0</horstretch>
+       <verstretch>0</verstretch>
+      </sizepolicy>
+     </property>
      <property name="title">
       <string>Open Project</string>
      </property>
      <property name="checkable">
       <bool>true</bool>
      </property>
-     <layout class="QHBoxLayout" name="horizontalLayout_2">
+     <layout class="QVBoxLayout" name="verticalLayout_2">
       <item>
-       <widget class="QPushButton" name="openProjectBtn">
-        <property name="text">
-         <string>Browse...</string>
+       <spacer name="verticalSpacer_6">
+        <property name="orientation">
+         <enum>Qt::Vertical</enum>
+        </property>
+        <property name="sizeHint" stdset="0">
+         <size>
+          <width>20</width>
+          <height>0</height>
+         </size>
```

Comment:
What's the point of this spacer?

Reply:
Setting the widgets' vertical SizePolicy to preferred means they match each others' height, but the spacers are there to make sure they don't expand any higher than the max height of the tallest in the hbox. It's necessary in some places, here I just added it for good measure since a stylesheet could easily make a button and a line edit different heights by default.